spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+require './spec/shared_context.rb'
+
+RSpec.describe BooksController, type: :controller do
+  describe 'GET #index' do
+    context 'When fetching all the books' do
+      let!(:books) { create_list(:book, 5) }
+
+      it 'responses with the books json' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          books, each_serializer: BookSerializer
+        ).to_json
+        expect(response.body.to_json) =~ JSON.parse(expected)
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    context 'When fetching a book' do
+      let!(:book) { create(:book) }
+
+      before do
+        get :show, params: { id: book.id }
+      end
+
+      it 'responses with the book json' do
+        expected = book.to_json
+        expect(response.body.to_json) =~ JSON.parse(expected)
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    context 'When a book does not exists' do
+    rescue ActiveRecord::RecordNotFound
```

Comment:
Here you are throwing the exception, I think you should do the same you did in the context 'when fetching book', create a book with an id, and do a get with a book_id that doesn't exists,

Reply:
I did it